test/shopify-cli/commands/create_test.rb

```diff
@@ -3,15 +3,65 @@
 module ShopifyCli
   module Commands
     class CreateTest < MiniTest::Test
+      include TestHelpers::Constants
+
+      def setup
+        super
+        load_cmd
+      end
+
       def test_without_arguments_calls_help
         @context.expects(:puts).with(ShopifyCli::Commands::Create.help)
-        run_cmd('create')
+        @cmd.call([], @cmd_name)
+      end
+
+      def test_with_create_app
+        Create::App.any_instance.expects(:call)
+          .with(['new-app'], 'app')
+        @cmd.call(['app', 'new-app'], @cmd_name)
+      end
+
+      def test_with_create_script_with_env_var_set
+        ENV.stubs(:[]).with('SCRIPTS_PLATFORM').returns('true')
+        load_cmd
+
+        Create::Script.any_instance.expects(:call)
+          .with(['discount', 'new-script'], 'script')
+        @cmd.call(['script', 'discount', 'new-script'], @cmd_name)
+      end
+
+      def test_create_script_is_hidden_unless_env_is_set
+        ENV.stubs(:[]).with('SCRIPTS_PLATFORM').returns(nil)
+        load_cmd
+
+        Create::Script.any_instance.expects(:call).never
+        @cmd.call(['script', 'discount', 'new-script'], @cmd_name)
+        assert_nil(@cmd.subcommand_registry.lookup_command('script').first)
+      end
+
+      def test_with_create_project_returns_warning
+        io = capture_io do
+          @cmd.call(['project', 'new-app'], @cmd_name)
+        end
+        assert_match('shopify create app', io.join)
+      end
+
+      private
+
+      def load_cmd
+        reload_class
+        @cmd = ShopifyCli::Commands::Create
+        @cmd.ctx = @context
+        @cmd_name = 'create'
       end
 
-      def test_with_project_calls_project
-        Create::Project.any_instance.expects(:call)
-          .with(['new-app'], 'project')
-        run_cmd('create project new-app')
+      def reload_class
+        ignore_constant_redefined_warnings do
+          ShopifyCli::Commands.send(:remove_const, :Create)
```

Comment:
I don't think I ever saw tests for this contextual command implementation but I don't like that it requires redefining constants in order to test properly. This has the potential to make testing even more complex and harder to understand.